syntax="proto2";

package doc_index_proto;

message Pair
{
  //每个分词结果的前闭后开区间
  required int32 beg = 1;
  required int32 end = 2;
};

//一个汉字：gdk ２个字节，　utf-8 大部分3个字节，得看具体的字，不确定

message DocInfo
{
  //一个文档对应的信息
  required uint64 doc_id = 1;
  required string title = 2;
  required string content = 3;
  required string jump_url = 4;
  required string show_url = 5;
  //以下两个字段,把标题和正文的分词结果先罗列好,
  //以备后面实现倒排的时候使用
  repeated Pair title_token = 6;
  repeated Pair content_token = 7;
};

//该词在文档中的权重信息
message Weight
{
  required uint64 doc_id = 1;
  required int32 weight = 2;
  required int32 title_count = 3;
  required int32 content_count = 4;
  //该词第一次在文档中出现的位置．
  //主要是为了后面生成描述
  required int32 first_pos = 5;
};

message KwdInfo
{
  //关键词
  required string key = 1;
  //倒排拉链
  repeated Weight weight = 2;
};

message Index
{
  repeated DocInfo forward_index = 1;
  repeated KwdInfo inverted_index = 2;
};

